content/post/different-ui-rendering-methods/index.md

```diff
@@ -0,0 +1,107 @@
+
+#### When to use SSR:
+
+- Developing content-focused websites, blogs, or e-commerce platforms where SEO is a priority.
+- Seeking faster initial page loads and improved perceived performance.
+- Catering to users with limited JavaScript support or slower devices.
+
+### 3. Static Site Generation (SSG):
+
+Static Site Generation involves pre-building the entire website during the build process, generating static HTML files for each page. SSG frameworks like Gatsby or Hugo, and Astro allow developers to create high-performance websites that can be served directly from a Content Delivery Network (CDN).
+
+![SSG](SSG.png)
+
+#### Advantages of SSG:
+
+- Exceptional performance: SSG-generated static HTML files can be cached and served directly, resulting in fast page rendering and reduced server load.
+- Cost-effective scalability: SSG sites can leverage CDNs, ensuring global scalability without the need for server-side infrastructure, leading to cost savings.
+- Enhanced security: With no server-side processing, the attack surface is minimized, making SSG sites more secure.
+
+#### When to use SSG:
+
+- Building content-heavy websites, blogs, or documentation sites where content updates are infrequent.
+- Requiring exceptional performance and scalability while reducing server-side complexity.
+- Prioritizing security by minimizing the attack surface.
+
+### 4. Universal Rendering (CSR + SSR):
+
+In universal rendering when the browser requests a URL with universal (server-side + client-side) rendering enabled, the server returns a fully rendered HTML page to the browser. Whether the page has been generated in advance and cached or is rendered on the fly, at some point, the browser runs the JavaScript code in a server environment, producing an HTML document. Users immediately get the content of our application, contrary to client-side rendering. This step is similar to traditional server-side rendering performed by PHP or Ruby applications.
+
+To not lose the benefits of the client-side rendering method, such as dynamic interfaces and page transitions, the Client (browser) loads the JavaScript code that runs on the Server in the background once the HTML document has been downloaded. The browser interprets it again (hence Universal rendering) and CSR takes control of the document and enables interactivity.
+
+Universal rendering allows a Nuxt application to provide quick page load times while preserving the benefits of client-side rendering. Furthermore, as the content is already present in the HTML document, crawlers can index it without the overhead.
+
+### Out Projects
+
+We have several web project to render. Some of them require high interactivity, some of then need SEO and others are content heavy. Depends on the project requirement to use different rendering method in our project. Here is a table that shows in which project we use which UI rending methods.
```

Comment:
Change the first sentence to "Here at Appscode, we have developed several web applications that utilize the different rendering techniques mentioned here."